packages/api/src/providers/google-tasks.ts

```diff
@@ -0,0 +1,97 @@
+import GoogleTasks from "@repo/google-tasks";
+import { TaskCollection, TaskCollectionWithTasks, Task, TaskProvider } from "./interfaces";
+import { ProviderError } from "./utils";
+import { parseGoogleTask, toGoogleTask } from "./google-calendar/utils";
+
+interface GoogleTasksProviderOptions {
+  accessToken: string;
+  accountId: string;
+}
+
+export class GoogleTasksProvider implements TaskProvider {
+  private static readonly TASK_LIST_ID_REGEX = /lists\/(.*?)\/tasks/;
+  public readonly providerId = "google" as const;
+  public readonly accountId: string;
+  private client: GoogleTasks;
+
+  constructor({ accessToken, accountId }: GoogleTasksProviderOptions) {
+    this.accountId = accountId;
+    this.client = new GoogleTasks({
+      accessToken,
+    });
+  }
+
+  async taskCollections(): Promise<TaskCollection[]> {
+    return this.withErrorHandler("get taskCollections", async () => {
+      const { items: taskCollections } = await this.client.tasks.v1.users.me.lists.list();
+      if (!taskCollections) return [];
+
+      return taskCollections
+        .filter((taskCollection) => taskCollection.id && taskCollection.title)
+        .map((taskCollection) => ({
+          id: taskCollection.id!,
+          providerId: "google",
+          title: taskCollection.title,
+          updated: taskCollection.updated,
+        }));
+    });
+  }
+  async tasks(): Promise<TaskCollectionWithTasks[]> {
+    const { items: taskCollections } = await this.client.tasks.v1.users.me.lists.list();
+
+    const results = await Promise.all(
+      taskCollections?.map(async (taskCollection) => {
+        const { items: tasks } = await this.client.tasks.v1.lists.tasks.list(taskCollection.id!);
+        return {
+          taskCollectionId: taskCollection.id!,
+          taskCollectionName: taskCollection.title!,
+          providerId: "google",
+          tasks: tasks?.map((task) => parseGoogleTask({ task, collectionId: taskCollection.id!, accountId: this.accountId })) ?? []
+        };
+      }) ?? []
+    );
+
+    return results;
+  }
+
+  createTask(task: Omit<Task, "id">): Promise<Task> {
+    return this.withErrorHandler("create task", async () => {
+      const createdTask = await this.client.tasks.v1.lists.tasks.create(task.taskCollectionId, task);
+      return parseGoogleTask({ task: createdTask, collectionId: task.taskCollectionId, accountId: this.accountId });
+    });
+  }
+
+  tasksForTaskCollection(taskCollectionId: string): Promise<Task[]> {
+    return this.withErrorHandler("get tasks for taskCollection", async () => {
```

Comment:
change all of them to the function name for consistency with the calendar providers 
`return this.withErrorHandler("tasksForTaskCollection", async () => {`